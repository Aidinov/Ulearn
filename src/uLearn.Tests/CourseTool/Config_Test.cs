using System.IO;
using ApprovalTests;
using ApprovalTests.Reporters;
using NUnit.Framework;
using Ulearn.Common.Extensions;

namespace uLearn.CourseTool
{
	[TestFixture]
	public class Config_Test
	{
		[Test]
		[UseReporter(typeof(DiffReporter))]
		public void ConfigTemplate_IsOk()
		{
			var path = Path.Combine(TestContext.CurrentContext.TestDirectory, "templates", "config.xml");
			var config = new FileInfo(path).DeserializeXml<Config>();
			config.IgnoredUlearnSlides = new[] { "1", "2" };
			Approvals.VerifyXml(config.XmlSerialize());
		}
	}
}