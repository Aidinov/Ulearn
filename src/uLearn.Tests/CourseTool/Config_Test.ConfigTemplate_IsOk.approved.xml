<Config>
  <Organization>UrFU</Organization>
  <CourseNumber>IMCS.b.CS-0093</CourseNumber>
  <CourseRun>2015</CourseRun>
  <LtiId>edx</LtiId>
  <ULearnCourseId>BasicProgramming</ULearnCourseId>
  <Video>video.json</Video>
  <Profiles>
    <Profile name="default">
      <EdxStudioUrl>https://openedu.urfu.ru:8081</EdxStudioUrl>
      <UlearnUrl>https://ulearn.me</UlearnUrl>
    </Profile>
    <Profile name="dev">
      <EdxStudioUrl>http://192.168.33.10:18010</EdxStudioUrl>
      <UlearnUrl>https://localhost:44300</UlearnUrl>
    </Profile>
  </Profiles>
  <IgnoredUlearnSlides>
    <SlideId>1</SlideId>
    <SlideId>2</SlideId>
  </IgnoredUlearnSlides>
  <EmitSequentialsForInstructorNotes>false</EmitSequentialsForInstructorNotes>
</Config>